File: leetcode/code/leetcode_1/1.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        int[] result = new int[2];
        int r = 0;
        int len = nums.Length;
        int temp = -1;
        for(int i = 0; i < len; i++)
        {
            if((nums[i] <= target && target >= 0) || (target < 0 && nums[i] >= target))
            {
                temp = target - nums[i];
                result[0] = i;
                for(int j = i+1; j < len; j++)
                {
                    if(nums[j] == temp)
                    {
                        result[1] = j;
                        return result;
                    }
                }
            }
            
            
        }
        
        return result;
        
    }
}

File: leetcode/leetcode_15/1.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {        
       
        List<List<Integer>> result = new ArrayList<List<Integer>>();
        HashSet<Integer> set = new HashSet<Integer>();
        HashSet<String> r = new HashSet<String>();
        
        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++)
            {
                int sum = -1 * (nums[i] + nums[j]);
                if (set.contains(sum)) {
                    if (r.contains(sum + "_" + nums[i] + "_" + nums[j] + "_")) continue;
                    List<Integer> list = new ArrayList<Integer>();
                    list.add(nums[i]);
                    list.add(nums[j]);
                    list.add(sum);
                    Collections.sort(list);
                    
                    String y = "";
                    for(Integer k: list) {
                        y += k + "_";                        
                    }
                    if (r.contains(y)) continue;
                    r.add(y);
                    result.add(list);
                } //if                
                
            } // for
            
            set.add(nums[i]);
        } // for
        
        return result;   
    }

}